src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -575,9 +575,6 @@ public KubernetesClient connect() throws KubernetesAuthException, IOException {
      */
     private boolean addProvisionedSlave(@Nonnull PodTemplate template, @CheckForNull Label label, int scheduledCount) throws Exception {
         int containerCap = getContainerCap();
-        if (containerCap == 0) {
-            return false;
-        }
```

Comment:
So this is going beyond reverting #828; now you are making API server calls even when no caps are in place, which would be a performance regression.

I think this method instead needs to be altered in structure. First, if `containerCap` is not infinite, compute `allPods` and check its size. Then, if `template.instanceCap` is not infinite, compute `labelPods` and check its size. If neither of those returns `false` then you return `true` by default.

And the method should be renamed to clarify that it is not actually adding a provisioned agent, but checking whether we _may_ add a provisioned agent.

Reply:
Thank you sir for the check, I don't disagree. Looking into it.

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/866/commits/0375b352a1987873b933712bb851331f4459e181 should do it.